### ISSUE TITLE
Add "Further Information" resource links to all doc pages

### ISSUE THREAD

**Issue by markerikson** [reduxjs/redux#2589]
As seen in my [React/Redux links list](https://github.com/markerikson/react-redux-links), there's _tons_ of useful and relevant info about all things Redux online.  The docs themselves can only cover so much actual info, and most of it is focused on the basics.  

I try to end each of my blog posts with a "Further Information" section that includes links mentioned earlier in the page, and also has additional links to relevant articles and resources.  I'd love it if we could add this type of info to each of our existing docs pages.

As a starting point: the "Structuring Reducers" docs section has a page called "Prerequisite Concepts" that links to some selected articles on topics like immutable data and normalization.  However, the other pages in that section don't specifically reference those links.  A few of those links could be added to the bottom of each page in the "Structuring Reducers" section, as appropriate.

For reference, a few of my posts:

- [Idiomatic Redux: The Tao of Redux, Part 1 - Implementation and Intent](http://blog.isquaredsoftware.com/2017/05/idiomatic-redux-tao-of-redux-part-1/)
- [Practical Redux, Part 6: Connected Lists, Forms, and Performance](http://blog.isquaredsoftware.com/2017/01/practical-redux-part-6-connected-lists-forms-and-performance/)
- [Blogged Answers: Webpack HMR vs React-Hot-Loader](http://blog.isquaredsoftware.com/2017/08/blogged-answers-webpack-hmr-vs-rhl/)

**Comment by renatoselenica** [comment]
I would like to help out with this one.

**Comment by markerikson** [comment]
@renatoselenica : great, thanks!

As I said in the issue, a good place to start would be adding appropriate links to the end of the "Structuring Reducers" docs section pages.  I don't have a specific list of which links I want on which pages - I'll leave that up to you to come up with some initial suggestions.  Please feel free to either add some comments here with your ideas so that we can discuss them, or go ahead and file some PRs and we'll go from there.

**Comment by renatoselenica** [comment]
Ok then. I will file a PR tomorrow and go from there. Would have done it today, but a few things came up.

**Comment by renatoselenica** [comment]
Hello. I just wanted to say that I cannot work on this issue for a while. I deeply apologize but some things came up unexpectedly. If someone else wants to take over, they are more then welcome to. Sorry for causing trouble

**Comment by markerikson** [comment]
That's okay.  Thanks for volunteering, and please let us know if you have time to help with this or other issues in the future!

**Comment by bonusrk** [comment]
I can try to pick up this issue.
@markerikson As you said in first post, "Prerequisite Concepts" has a list of articles to read, and this list covers a bit almost all pages of  "Structuring Reducers" section.

Maybe it is not a bad idea to copy some of them to related pages (for example **Normalizing State Shape**) with some additional links.
Like this article on Medium and 2 chapters from your glorious **Practical Redux**.

[Normalizing Redux stores for maximum code reuse.](https://medium.com/@adamrackis/normalizing-redux-stores-for-maximum-code-reuse-ae6e3844ae95)
[Practical Redux, Part 1: Redux-ORM Basics](http://blog.isquaredsoftware.com/2016/10/practical-redux-part-1-redux-orm-basics/)
[Practical Redux, Part 2: Redux-ORM Concepts and Techniques](http://blog.isquaredsoftware.com/2016/10/practical-redux-part-2-redux-orm-concepts-and-techniques/)

Is it what you meant?

**Comment by markerikson** [comment]
@bonusrk : yeah, pretty much.  In other words, the "Prerequisite Concepts" page links to "stuff it would be good to know before reading the rest of Structuring Reducers".  But, if someone follows a link directly to the "Normalizing State Shape" page, they don't see any of those links.  So, putting several related links at the bottom of the page would be helpful for people who want to learn more on the topic.

**Comment by bonusrk** [comment]
@markerikson got it. Then i'll make a list like **"pages : links-to-place-there"**  and place them here for your approval. Of course with usage of your [React/Redux links list](https://github.com/markerikson/react-redux-links)  treasury :)

**Comment by markerikson** [comment]
@bonusrk : sounds good - let me know when you've got a list put together!

**Comment by bonusrk** [comment]
@markerikson  thats how it looks for now. I still digging your "links mine".
The problem is that most part of articles there is so cool, that i just cant pass through them and read them fully =)
And of course it is just for **Structuring Reducers** part

**Basic Reducer Structure**
_Idea_ : Give some basic information about Redux architecture, which starts from basic reducer structure  
***Links***

[10 Tips for Better Redux Architecture](https://medium.com/javascript-scene/10-tips-for-better-redux-architecture-69250425af44)

[Redux Best Practices](https://medium.com/lexical-labs-engineering/redux-best-practices-64d59775802e)

["Redux Best Practices?"](https://www.reddit.com/r/javascript/comments/3p9o0j/redux_best_practices/)



**Splitting Reducer Logic**
_Idea_: Give some best practice
***Links***
["Reducer composition without slicing state"](https://www.reddit.com/r/javascript/comments/42ey9e/redux_reducer_composition_without_slicing_state/)

[The Power of Higher-Order Reducers](http://slides.com/omnidan/hor#/)
[Reducer composition with Higher Order Reducers](https://medium.com/@mange_vibration/reducer-composition-with-higher-order-reducers-35c3977ed08f)


**Refactoring Reducers Example**
_nothing yet_


**Using combineReducers**
_Idea_: Some tips how and when  use combineReducer
***Links***
[4 Things for Efficient Redux](https://blog.spacelab.team/4-things-for-efficient-redux-17e87b4cd6cb)
[Taking Advantage of combineReducers](http://randycoulman.com/blog/2016/11/22/taking-advantage-of-combinereducers/)

**Beyond combineReducers**
_nothing here yet_

**Normalizing State Shape**
_Idea_ : Give more info about normalizing in general
***Links***
[Redux Normalizr: Improve your State Management](http://www.robinwieruch.de/the-soundcloud-client-in-react-redux-normalizr/)
[Normalizing Redux stores for maximum code reuse.](https://medium.com/@adamrackis/normalizing-redux-stores-for-maximum-code-reuse-ae6e3844ae95)
[Practical Redux, Part 1: Redux-ORM Basics](http://blog.isquaredsoftware.com/2016/10/practical-redux-part-1-redux-orm-basics/)
[Practical Redux, Part 2: Redux-ORM Concepts and Techniques](http://blog.isquaredsoftware.com/2016/10/practical-redux-part-2-redux-orm-concepts-and-techniques/)
[Advanced Redux Entity Normalization](https://medium.com/@dcousineau/advanced-redux-entity-normalization-f5f1fe2aefc5)
["Why are you supposed to normalize data in Redux?"](https://twitter.com/AdamRackis/status/847883670950219776)

Updating Normalized Data
_need investigation_

**Reusing Reducer Logic**
_Idea_ :Give some best practice about DRY inside redux
***Links***
[Reducer composition with Higher Order Reducers](https://medium.com/@mange_vibration/reducer-composition-with-higher-order-reducers-35c3977ed08f)
[Higher Order Reducers - It just sounds scary](https://medium.com/@danielkagan/high-order-reducers-it-just-sounds-scary-2b9e5dbfc705)


**Immutable Update Patterns**
_Idea_ : Give additional info about immutability and it's usage inside reducers
***Links***
[Modifying a Leaf of the Redux State Tree](http://cmichel.io/modifying-a-leaf-of-the-redux-state-tree/)


Initializing State

**Comment by bonusrk** [comment]
@markerikson Does it looks reasonable to be added?

**Comment by markerikson** [comment]
@bonusrk : a few initial thoughts:

- I don't think that the "architecture"-related links would really be appropriate for the "Basic Reducer Structure" page.  Instead, I'd suggest these links:
  - https://redux.js.org/docs/basics/Reducers.html
  - https://redux.js.org/docs/introduction/CoreConcepts.html
  - http://jamesknelson.com/5-types-react-application-state/
- Looks like some of the links in my list are dead (eep!).  In particular, the "4 things for efficient Redux" post is now at https://medium.com/@iJaniashvili/4-things-for-efficient-redux-17e87b4cd6cb .
- For "Immutable Update Patterns", I'd add:
  - https://daveceddia.com/immutable-updates-react-redux/
  - http://blog.jakoblind.no/most-common-redux-mistake/

Also, look at the ["Prerequisite Concepts" page ](https://redux.js.org/docs/recipes/reducers/PrerequisiteConcepts.html), and see which of those links would be appropriate to also have at the bottom of different pages (like some of the immutable data links).

**Comment by bonusrk** [comment]
@markerikson  got it. will dig that way.
